pkg/controller/cluster/cluster_controller.go

```diff
@@ -127,13 +128,32 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 			klog.Errorf("Error deleting cluster object %v; %v", name, err)
 			return reconcile.Result{}, err
 		}
+
 		// Remove finalizer on successful deletion.
 		klog.Infof("cluster object %v deletion successful, removing finalizer.", name)
-		cluster.ObjectMeta.Finalizers = util.Filter(cluster.ObjectMeta.Finalizers, clusterv1.ClusterFinalizer)
-		if err := r.Client.Update(context.Background(), cluster); err != nil {
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// It's possible the actuator's Delete call modified the cluster. We can't guarantee that every provider
+			// updated the in memory cluster object with the latest copy of the cluster, so try to get a fresh copy.
+			//
+			// Note, because the get from the client is a cached read from the shared informer's cache, there's still a
+			// chance this could be a stale read.
+			//
+			// Note 2, this is not a Patch call because the version of controller-runtime in the release-0.1 branch
+			// does not support patching.
+			err := r.Get(context.Background(), request.NamespacedName, cluster)
```

Comment:
Hi @ncdc,

> the version of controller-runtime in the release-0.1 branch does not support patching.

What do you mean? Both CAPA and CAPV patch the cluster object using a JSON patch.

Reply:
controller-runtime's generic client doesn't have Patch support